engines/account/app/services/account/signup_service.rb

```diff
@@ -23,7 +20,7 @@ def create!(email)
     # @return [Account::User]
     def complete!(signup, options)
       ::ActiveRecord::Base.transaction do
-        user = ::Account::User.create(
+        user = ::Account::User.create!(
           email: signup.email,
           username: options[:username],
           password: options[:password],
```

Comment:
What we need to do instead is handle it in the controller. This is how transaction block are supposed to work. What you suggest is implementing transactions by hand.